misk/src/main/kotlin/misk/web/ServletHttpCall.kt

```diff
@@ -25,8 +25,16 @@ internal data class ServletHttpCall(
   var responseBody: BufferedSink? = null,
   var webSocket: WebSocket? = null
 ) : HttpCall {
+  private var _actualStatusCode: Int? = null
 
   override var statusCode: Int
+    get() = _actualStatusCode ?: upstreamResponse.statusCode
+    set(value) {
+      _actualStatusCode = value
+      upstreamResponse.statusCode = value
+    }
+
+  override var networkStatusCode: Int
     get() = upstreamResponse.statusCode
     set(value) {
       upstreamResponse.statusCode = value
```

Comment:
there’s a bit of a hazard here where you gotta set this one last. Ugh!

Maybe add a function and make `networkStatusCode` a val?

```
fun setStatusCodes(statusCode: Int, networkStatusCode: Int) { ... }
```

Reply:
Yeah, good call